File: src/main/java/rk/cinema/model/Seat.java
package rk.cinema.model;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Relation;

@MappedEntity
public record Seat(
        @Id
        @GeneratedValue
        Long id,

        @NonNull
        String row,

        @NonNull
        Integer number,

        @Relation(Relation.Kind.MANY_TO_ONE)
        @NonNull
        Hall hall
) {
}

File: src/main/kotlin/rk/cinema/model/Seat.kt
package rk.cinema.model

import io.micronaut.core.annotation.NonNull
import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.Relation

@MappedEntity
data class Seat(
    @GeneratedValue @param:Id val id: Long,
    val row: @NonNull String,
    val number: @NonNull Int,
    @Relation(Relation.Kind.MANY_TO_ONE) val hall: @NonNull Hall,
)
